xmr-btc/tests/harness/transport.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use tokio::{
+    stream::StreamExt,
+    sync::mpsc::{Receiver, Sender},
+};
+use xmr_btc::{
+    alice, bob,
+    transport::{Receive, Send},
+};
+
+#[derive(Debug)]
+pub struct Transport<SendMsg, RecvMsg> {
+    pub sender: Sender<SendMsg>,
+    pub receiver: Receiver<RecvMsg>,
+}
+
+#[async_trait]
+impl Send<alice::Message> for Transport<alice::Message, bob::Message> {
+    async fn send_message(&mut self, message: alice::Message) -> Result<()> {
+        let _ = self
+            .sender
+            .send(message)
+            .await
+            .map_err(|_| anyhow!("failed to send message"))?;
+        Ok(())
+    }
+}
```

Comment:
Should be able to implement this generically no?

```suggestion
#[async_trait]
impl<A, B> Send<A> for Transport<A, B> {
    async fn send_message(&mut self, message: A) -> Result<()> {
        let _ = self
            .sender
            .send(message)
            .await
            .map_err(|_| anyhow!("failed to send message"))?;
        Ok(())
    }
}
```

Reply:
That would save you two of these impl blocks.

Reply:
Good pick!